crates/matrix-sdk-common/src/linked_chunk/mod.rs

```diff
@@ -190,6 +190,40 @@ impl<const CAP: usize, Item, Gap> Ends<CAP, Item, Gap> {
             chunk = chunk.previous_mut()?;
         }
     }
+
+    /// Drop all chunks, and re-create the first one.
+    fn clear(&mut self) {
+        // Loop over all chunks, from the last to the first chunk, and drop them.
+        {
+            // Take the latest chunk.
+            let mut current_chunk_ptr = self.last.or(Some(self.first));
+
+            // As long as we have another chunk…
+            while let Some(chunk_ptr) = current_chunk_ptr {
+                // Disconnect the chunk by updating `previous_chunk.next` pointer.
+                let previous_ptr = unsafe { chunk_ptr.as_ref() }.previous;
+
+                if let Some(mut previous_ptr) = previous_ptr {
+                    unsafe { previous_ptr.as_mut() }.next = None;
+                }
```

Comment:
Do we need to do that unlinking? I fail to see if it achieves anything, considering we're about to drop the chunk on the next line…

Reply:
Indeed 🤔, it seems useless. After further investigation, I don't see any proof it is useful today. Let's remove that.